code/ship/ship.cpp

```diff
@@ -14809,12 +14809,32 @@ int ship_find_subsys(ship *sp, const char *ss_name)
 	return -1;
 }
 
+// an optimization of the below function that skips the subsystem iteration if we don't care about the exact number
+bool ship_subsystems_blown(const ship* shipp, int type, bool skip_dying_check)
+{
+	Assert ( (type >= 0) && (type < SUBSYSTEM_MAX) );
```

Comment:
This PR really is simple.  Just change this to an assertion, and I think we're good to go.